File: Data Structures & Algorithms/baseball-game/submission-0.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []
        for symbol in operations:
            if symbol == "+":
                stack.append(stack[-1] + stack[-2])
            elif symbol == "D":
                stack.append(2 * stack[-1])
            elif symbol == "C":
                stack.pop()
            else:
                stack.append(int(symbol))
        return sum(stack)


        

File: Data Structures & Algorithms/validate-parentheses/submission-0.java
class Solution {
    public boolean isValid(String s) {

        Stack<Character> valid = new Stack<>();

        Map<Character, Character> closeOpen = new HashMap<>();

        closeOpen.put(')', '(');
        closeOpen.put('}', '{');
        closeOpen.put(']', '[');

        for (char c : s.toCharArray()) {
            if (closeOpen.containsKey(c)) {
                if (!valid.isEmpty() && valid.peek() == closeOpen.get(c)) {
                    valid.pop();
                } else {
                    return false;
                }
            }
            else {
                valid.push(c);
            }
        }

        return valid.isEmpty();


        
    }
}
